src/entity/user/UserStreakAction.ts

```diff
@@ -0,0 +1,25 @@
+import { Entity, JoinColumn, OneToOne, PrimaryColumn } from 'typeorm';
+import { User } from './User';
+
+export enum UserStreakActionType {
+  Recover = 'recover',
+}
+
+@Entity()
+export class UserStreakAction {
+  @PrimaryColumn({ length: 36 })
+  userId: string;
+
+  @PrimaryColumn({ type: 'text' })
+  type: UserStreakActionType;
+
+  @PrimaryColumn({ type: 'timestamptz', default: () => 'now()' })
+  createdAt: Date;
+
+  @OneToOne(() => User, (user) => user.streak, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  @JoinColumn({ name: 'userId' })
```

Comment:
You should not need join column right? Typeorm usually infers automatically?

Reply:
Just copy-pasted it from UserStreak entity, but we can try.

Reply:
For some reason, the typeorm did not create the right reference for creating an FK.

Reply:
I see now, `UserStreak` PK is `userId` and it looks for `id` or `userStreakId`. Good then 👍